src/Client/Connection.cpp

```diff
@@ -288,6 +309,19 @@ void Connection::forceConnected(const ConnectionTimeouts & timeouts)
     }
 }
 
+#if USE_SSL
+void Connection::sendCluster()
+{
+    pcg64_fast rng(randomSeed());
+    UInt64 rand = rng();
+
+    salt = encodeSHA256(&rand, sizeof(rand));
+
+    writeStringBinary(cluster, *out);
+    writeStringBinary(salt, *out);
+}
```

Comment:
I think it's useful to add something like version in these fields.
So if in the future we have to change something in the hashing in future we will able to do it easily.

Reply:
>So if in the future we have to change something in the hashing we will able to do it easily.

But server revision will be enough for this, no?
Adding one more version here does not looks good to me.

Reply:
ok